docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/entrypoints/addl-cancer-types-landing-page/components/cgdp-cancer-type-az-list.ts

```diff
@@ -0,0 +1,187 @@
+		componentType: 'Raw HTML',
+		pageRows,
+		pageRowIndex,
+		pageRowCols,
+		pageRowColIndex,
+		containerItems,
+		containerItemIndex,
+		componentTheme: 'Not Defined',
+		componentVariant: 'Accordion',
+		title: 'Cancer Types A-Z Accordion',
+		linkArea,
+		linkText: linkText.slice(0, 50),
+		linkType,
+		totalLinks: allLinks.length,
+		linkPosition,
+		accordionExpands,
+		accordionCollapses,
+		sectionsExpandedAtClick,
+		accordionAction: 'Link Click',
+		accordionFirstInteraction: isFirstInteraction,
+	});
+};
+
+/**
+ * Handles click events in accordion.
+ * @param {Event} evt
+ */
+const accordionLinkClickHandler = (evt: Event): void => {
+	const target = evt.currentTarget as HTMLElement;
+	accordionClickTracker(target, accordionFirstInteraction);
+	updateAccordionFirstInteraction();
+};
+
+/**
+ * Handles expand/collapse events in accordion.
+ * @param {Event} evt
+ */
+const accordionExpandCollapseHandler = (evt: Event): void => {
+	accordionExpandCollapseTracker(evt, accordionFirstInteraction);
+	updateAccordionFirstInteraction();
+};
+
+/**
+ * Counter for Expands.
+ * @param {Event} evt
+ */
+const accordionExpandCollapseTracker = (
+	evt: Event,
+	isFirstInteraction: boolean
+) => {
+	const target = evt.currentTarget as HTMLElement;
+
+	if (target.getAttribute('aria-expanded') === 'false') {
+		accordionCollapses += 1;
+	} else {
+		accordionExpands += 1;
+	}
+	updateAccordionFirstInteraction();
+
+	const { pageRows, pageRowIndex, pageRowCols, pageRowColIndex } =
+		getLandingRowsAndColsInfo(target);
+
+	const accordionSection = target.closest('.usa-accordion');
+	const allContainers =
+		accordionSection !== null
+			? Array.from(accordionSection.querySelectorAll('.usa-accordion__heading'))
+			: [];
+	const containerItems = allContainers.length;
+	const currentContainer = target.closest('.usa-accordion__heading');
+	const containerItemIndex =
+		currentContainer !== null ? allContainers.indexOf(currentContainer) + 1 : 0;
+
+	const linkText = target.textContent ? target.textContent.trim() : '_ERROR_';
+
+	trackOther(
+		`LP:RawHTMLAccordion:ExpandCollapse`,
+		`LP:RawHTMLAccordion:ExpandCollapse`,
+		{
+			location: 'Body',
+			componentType: 'Raw HTML',
+			pageRows,
+			pageRowIndex,
+			pageRowCols,
+			pageRowColIndex,
+			containerItems,
+			containerItemIndex,
+			componentTheme: 'Not Defined',
+			componentVariant: 'Accordion',
+			title: 'Cancer Types A-Z Accordion',
+			linkArea: linkText,
+			linkText,
+			linkType: 'Anchor',
+			accordionExpands,
+			accordionCollapses,
+			accordionAction: 'Link Click',
+			accordionFirstInteraction: isFirstInteraction,
+		}
+	);
+};
+
+/**
+ * Updates the first interaction variable.
+ */
+const updateAccordionFirstInteraction = (): void => {
+	if (accordionFirstInteraction) {
+		accordionFirstInteraction = false;
+	}
+};
+
+/**
+ * Wires up component per cgdp requirements.
+ */
+const initialize = (): void => {
+	const accordionAnchorTags = document.querySelectorAll(
```

Comment:
You really should have some other identifier for the a-z list here. I am not saying there will be other accordions on the page ever, but if we did something like update the footer to use usa-accordion like THEY use for expand/collapse this a-z specific code is going to attach events to those items.

Remember, the A-Z list is displaying its contents using a `usa-accordion` it is not *the* accordion.